roles/ceph-validate/tasks/check_iscsi.yml

```diff
@@ -1,21 +1,17 @@
 ---
-- name: set_fact use_new_ceph_iscsi package or old ceph-iscsi-config/cli
-  set_fact:
-    use_new_ceph_iscsi: "{{ (gateway_ip_list | default('0.0.0.0') == '0.0.0.0' and gateway_iqn | default('') | length == 0 and client_connections | default({}) | length == 0 and rbd_devices | default({}) | length == 0) | bool | ternary(true, false) }}"
-
 - name: make sure gateway_ip_list is configured
   fail:
     msg: "you must set a list of IPs (comma separated) for gateway_ip_list"
   when:
-    - "gateway_ip_list | default('0.0.0.0') == '0.0.0.0'"
+    - gateway_ip_list == '0.0.0.0'
     - not containerized_deployment | bool
     - not use_new_ceph_iscsi | bool
 
 - name: make sure gateway_iqn is configured
   fail:
     msg: "you must set a iqn for the iSCSI target"
   when:
-    - "gateway_iqn | default('') | length == 0"
+    - gateway_iqn | length == 0
     - not containerized_deployment | bool
     - not use_new_ceph_iscsi | bool
```

Comment:
I think with the change in this patch, we can also drop the default() for client_connections a little bit below this chunk. It was in the original check_iscsi.yml:

  with_items: "{{ client_connections | default({})  }}"

Do you want me to submit a patch over your patch for that?

Reply:
good catch. I've modified my commit, thanks!